src/features/marketplace/process/commands/delist-item/store-delist-item-command.ts

```diff
@@ -0,0 +1,27 @@
+import ItemService from '../../../../leda-nft/services/item.service';
+import MarketplaceError from '../../enums/marketplace-error.enum';
+import ICommand from '../../interfaces/command.interface';
+import MarketplaceState from '../../types/marketplace-state';
+
+export default class StoreDelistItemCommand implements ICommand<MarketplaceState> {
+  private readonly itemService: ItemService;
+
+  constructor(_itemService: ItemService) {
+    this.itemService = _itemService;
+  }
+
+  async execute(state: MarketplaceState): Promise<MarketplaceState> {
+    if (!state.itemId) return { ...state, error: MarketplaceError.RequiredItemId };
+    if (!state.marketplaceEvent) return { ...state, error: MarketplaceError.RequiredListEvent };
```

Comment:
You are missing the ownerAddress check. The marketplaceEvent is not required here